src/test/java/integration/CollectionMethodsTest.java

```diff
@@ -209,19 +212,53 @@ public void canIterateCollection_withIterator() {
   @Test
   public void canIterateCollection_withListIterator() {
     ListIterator<SelenideElement> it = $$("[name=domain] option").listIterator(3);
-    assertTrue(it.hasNext()); 
-    assertTrue(it.hasPrevious()); 
+    assertTrue(it.hasNext());
+    assertTrue(it.hasPrevious());
     it.previous().shouldHave(text("@rusmail.ru"));
-    
-    assertTrue(it.hasPrevious()); 
+
+    assertTrue(it.hasPrevious());
     it.previous().shouldHave(text("@myrambler.ru"));
-    
-    assertTrue(it.hasPrevious()); 
+
+    assertTrue(it.hasPrevious());
     it.previous().shouldHave(text("@livemail.ru"));
-  
+
     assertFalse(it.hasPrevious());
-    
+
     it.next().shouldHave(text("@livemail.ru"));
     assertTrue(it.hasPrevious());
   }
+
+  @Test
+  public void canGetFirstNElements() {
+    ElementsCollection collection =  $$x("//select[@name='domain']/option");
+    collection.first(2).shouldHaveSize(2);
+    collection.first(10).shouldHaveSize(collection.size());
+
+    List<String> regularSublist = $$x("//select[@name='domain']/option").stream()
+            .map(SelenideElement::getText)
+            .collect(Collectors.toList()).subList(0, 2);
+
+    List<String> selenideSublist = collection.first(2).stream()
+            .map(SelenideElement::getText)
+            .collect(Collectors.toList());
+
+    Assert.assertEquals(regularSublist, selenideSublist);
+  }
+
+  @Test
+  public void canGetLastNElements() {
+    ElementsCollection collection =  $$x("//select[@name='domain']/option");
+    collection.last(2).shouldHaveSize(2);
+    collection.last(10).shouldHaveSize(collection.size());
+
+    List<String> regularSublist = $$x("//select[@name='domain']/option").stream()
+            .map(SelenideElement::getText)
+            .collect(Collectors.toList()).subList(2, collection.size());
+
+    List<String> selenideSublist = collection.last(2).stream()
+            .map(SelenideElement::getText)
+            .collect(Collectors.toList());
+
+    Assert.assertEquals(regularSublist, selenideSublist);
```

Comment:
In addition - just a few tests are not enough. It does make sense to check error messages for such "sub-collections"

Reply:
What error messages do you have in mind?

Reply:
In Selenide we have kind of "informative error messages" in cases when the "should" method failed, or some operation on collection element failed. In such cases the message should be really informative. The user should understand from the error message what was the collection that he tried to deal with. 

Just run:

$$(".item").last(3).shouldHaveSize(4)
And check the error message. It should be informative. But probably it will not:)
And we should have tests for such cases too.